sdk/extensions/Azure.Extensions.AspNetCore.DataProtection.Keys/src/DecryptorTypeForwardingActivator.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.DataProtection.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+#pragma warning disable AZC0001
+namespace Microsoft.AspNetCore.DataProtection
+#pragma warning restore AZC0001
+{
+    /// <summary>
+    /// This type is a copy of https://github.com/dotnet/aspnetcore/blob/e1bf38ccaf2a98f95e48bf22b8b76d996a0c33ea/src/DataProtection/DataProtection/test/TypeForwardingActivatorTests.cs
+    /// with addition of AzureKeyVaultXmlDecryptor forwarding logic.
+    /// </summary>
+    internal class DecryptorTypeForwardingActivator : IActivator
+    {
+        private readonly IServiceProvider _services;
+        private const string OldNamespace = "Microsoft.AspNet.DataProtection";
+        private const string CurrentNamespace = "Microsoft.AspNetCore.DataProtection";
+        private const string CurrentAzureNamespace = "Azure.Extensions.AspNetCore.DataProtection.Keys";
+
+        private const string OldDecryptor = "Microsoft.AspNetCore.DataProtection.AzureKeyVault.AzureKeyVaultXmlDecryptor";
+        private const string OldDecryptorAssembly = "Microsoft.AspNetCore.DataProtection.AzureKeyVault";
+        private const string NewDecryptor = "Azure.Extensions.AspNetCore.DataProtection.Keys.AzureKeyVaultXmlDecryptor";
+        private const string NewDecryptorAssembly = "Azure.Extensions.AspNetCore.DataProtection.Keys";
+
+        private readonly ILogger _logger;
+        private static readonly Regex _versionPattern = new Regex(@",\s?Version=[0-9]+(\.[0-9]+){0,3}", RegexOptions.Compiled, TimeSpan.FromSeconds(2));
```

Comment:
Can't you just use an `AssemblyName` construct to parse this than more-expensive regex?

Reply:
AssemblyName would throw when `The referenced assembly could not be found, or could not be loaded.`. Regex was decided to be safer from an error handling perspective and making sure we don't throw when we shouldn't.

Reply:
My mistake. I incorrectly remembered it only parsed the name.